ampl-converter/src/main/java/com/powsybl/ampl/converter/AmplNetworkWriter.java

```diff
@@ -1339,12 +1448,23 @@ private void writeShunts(AmplExportContext context) throws IOException {
                 int vlNum = mapper.getInt(AmplSubset.VOLTAGE_LEVEL, t.getVoltageLevel().getId());
                 double vb = t.getVoltageLevel().getNominalV();
                 double zb = vb * vb / AmplConstants.SB;
-                double b1 = 0;
-                double b2 = sc.getModel(ShuntCompensatorLinearModel.class).getBPerSection() * sc.getMaximumSectionCount() * zb;
+                double b1;
+                double b2;
+                int points;
+                int sectionCount = 1;
+                if (sc.getModelType() == ShuntCompensatorModelType.NON_LINEAR) {
+                    b1 = sc.getModel(ShuntCompensatorNonLinearModel.class).getAllSections().get(sc.getSectionCount() - 1).getB() * zb;
+                    b2 = sc.getB() * zb;
+                    points = 0;
```

Comment:
Why the code is different for the two models: the getMaximumSectionCount is defined in the ShuntCompensator interface, so I guess that points should have the same value.

Reply:
Because at this stage, I am making an approximation for non linear shunts. We have to model non linear shunt as rtc and ptc and I think that we have to introduce a new format in the ampl converter. @MioRtia will help me for that but maybe later.